From b7f5ebbffc8d32ac726cc1875c4e6f02c4359e99 Mon Sep 17 00:00:00 2001
From: Chris Lamb <chris@chris-lamb.co.uk>
Date: Wed, 12 Apr 2017 19:50:27 +0100
Subject: [PATCH] Make the .pc output reproducible

Whilst working on the Reproducible Builds effort [0], we noticed
that sunpinyin could not be built reproducibly due to iterating over
the filesystem in a non-deterministic ordering

 [0] https://reproducible-builds.org/

Author:    Chris Lamb <chris@chris-lamb.co.uk>
Signed-off-by: Chris Lamb <chris@chris-lamb.co.uk>

The patch was imported from the github git server
(git@github.com:sunpinyin/sunpinyin) as of commit id
01b388134f657b73fa95e99999907678e9623870
---
 SConstruct | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 01b5cca..412d0d4 100644
--- a/SConstruct
+++ b/SConstruct
@@ -429,7 +429,7 @@ env.Substfile('sunpinyin-2.0.pc.in', SUBST_DICT={
     '@VERSION@': version,
     '@CFLAGS@': reduce(lambda a, b: a + ' ' + b,
                        map(lambda x: '-I$${includedir}' + x[3:],
-                           allinc())),
+                           sorted(allinc()))),
 })
 
 libname_default = '%ssunpinyin%s' % (env.subst('${SHLIBPREFIX}'),
-- 
1.9.0

